browser/todo-list.md

```diff
@@ -3,291 +3,248 @@
-| `components/Profile.js` | Application data storage and user sign out. |
+    ```
 
-## Task 2: Sign into the application
+    This component loads the [`UserSession`](https://blockstack.github.io/blockstack.js/classes/usersession.html) module from a second Blockstack library called [blockstack.js](https://github.com/blockstack/blockstack.js/), which complements Blockstack Connect by providing an API for many protocol-level operations, such as for authentication and storage.
 
-The example application runs in a node server on your local host. In the this section, you start the application and interact with it.
+    ```js
 
-1. Make sure you are in the root of the code base.
+    import { UserSession } from 'blockstack';
+    import { appConfig } from '../assets/constants'
 
-   ```bash
-    $ pwd 
-    /Users/meepers/repos/blockstack-todos
-   ```
-   
-   This path will be different for you, but double-check the last part to ensure that you're in the directory into which you cloned and in which you ran `npm install`.
+    // ...
 
-2. Start the application in your local environment.
+    const userSession = new UserSession({ appConfig })
 
-    ```bash
-    $ npm run start
     ```
-    
-    You should see output similar to the following:
 
-    ```bash
-      Compiled successfully!
+    This module handles user session operations and is initiated using the [`appConfig`](https://github.com/blockstack/blockstack-todos/blob/master/src/assets/constants.js#L3) object, which contains an array of [scopes](/develop/overview_auth.html#scopes) that indicate just what permissions to grant during authentication:
 
-      You can now view bs-todo in the browser.
+    ```js
+    export const appConfig = new AppConfig(['store_write', 'publish_data'])
+    ```
 
-        http://127.0.0.1:3000/
+    The `appDetails` and `userSession` objects are joined by the callback function [`finished`](https://github.com/blockstack/blockstack-todos/blob/master/src/components/App.jsx#L31) in configuring Blockstack Connect for authentication with the `authOptions` object:
 
-      Note that the development build is not optimized.
-      To create a production build, use npm run build.
-      ```
+    ```js
+    finished: ({ userSession }) => {
+      this.setState({ userData: userSession.loadUserData() });
+    }
 
-2. Open your local browser to the `http://localhost:3000` URL.
-  
-    You should see a simple application:
+    ```
 
-    ![](images/todo-sign-in.png)
+    This function simply saves data about the user into the app's state upon authentication.
 
- 3. Choose **Sign In with Blockstack**.
+    Further down in the component we see in [`componentDidMount`](https://github.com/blockstack/blockstack-todos/blob/master/src/components/App.jsx#L46) that it checks upon mount to either process completion of authentication with `userSession.handlePendingSignIn()` or otherwise load session data into app state as above with `userSession.isUserSignedIn()`:
```

Comment:
I'd point out that it loads session data in case a user is already logged in. Pointing this out might help the reader's understanding as to why there needs to be the same line of code (`this.setState({ userData: userSession.loadUserData() })`) twice in the code. One for the button action and one for a page load/refresh/mount